docs/contributing/code-quality-principles.md

```diff
@@ -91,8 +91,8 @@ This leads to safer code.
 
 ### Reasons
 
-We use scalar types (`int`, `float`, `string`, `bool`) as much as possible in new code.
-Using scalar types without strict mode leads to automatic type cast that can be even harmful (eg. input string `""` is cast to integer `0`).
+We use scalar type hints (`int`, `float`, `string`, `bool`) as much as possible in new code.
```

Comment:
I prefer to keep term `scalar types` as it is used in https://wiki.php.net/rfc/scalar_type_hints_v5

Anyway if you thing `scalar type hints` is more obvious to reader, please fix the term in the whole article

Reply:
The term that is used there is *scalar type declarations*.

I think that saying we use *scalar types* as much as possible means that we try to avoid arrays and objects.

Reply:
Yes, that was my feeling too - I believe there is a difference between "scalar types" (ie. integers, booleans, ...) and "scalar type hints"

Reply:
However, I don't see any meaningful difference between *"scalar type declarations"* and *"scalar type hints"*, so we can keep the terminology from [wiki.php.net](https://wiki.php.net/rfc/scalar_type_hints_v5).

Reply:
...there are two more occurrences of "scalar types" in the article, both of them are fine by me...